File: trunk/Myoushu/Samples/MyoushuJava/SampleCubes/src/SampleCubes.cpp
/**
 * @file SampleCubes.cpp
 * @author ME Chamberlain
 * @date June 2008
 */

#include "KernelManager.h"
#include "Constants.h"

#include "SampleCubes.h"
#include "SampleCubesGameTask.h"

SampleCubes::SampleCubes()
{
}

SampleCubes::~SampleCubes()
{
}

void SampleCubes::start(int argc, char **argv)
{
	SampleCubesGameTask *gameTask;

	// Initialize the Myoushu engine
	try
	{
		this->main(argc, argv);
		this->init();
	}
	catch (...)
	{
		return;
	}

	// Create and add the SampleCubesGameTask to the Kernel
	gameTask = new SampleCubesGameTask(Myoushu::Constants::GAME_TASK_PRIORITY, "SampleCubesGameTask"
		, Myoushu::Constants::GAME_TASK_EXECUTION_INTERVAL, Myoushu::Constants::GAME_TASK_ITERATION_LIMIT);
	Myoushu::KernelManager::getSingleton().addTask(gameTask);
	Myoushu::KernelManager::getSingleton().initAllTasks();

	this->run();
}

File: Myoushu/Samples/MyoushuJava/SampleCubes/src/SampleCubes.java
/**
 * @file SampleCubes.Java
 * @author ME Chamberlain
 * @date January 2009
 */
 
import MyoushuJava.*;
 
public class SampleCubes extends Application
{

	public SampleCubes()
	{
		super();
	}
	
	public void start()
	{
		SampleCubesGameTask gameTask;

		// Initialize the Myoushu engine
		try
		{
			this.main();
            loadPlugin("MyoushuJava");
            loadPlugin("myoushuDotScene");
			init();
		}
		catch (java.lang.Exception e)
		{
			return;
		}

        // Run the GUIInit.lua script
        ScriptManager.getSingleton().executeScriptFile("GUIInit.lua", Constants.getRG_GUI());

		// Create and add the SampleCubesGameTask to the Kernel
		gameTask = new SampleCubesGameTask(Constants.getGAME_TASK_PRIORITY(), "SampleCubesGameTask"
			, Constants.getGAME_TASK_EXECUTION_INTERVAL(), Constants.getGAME_TASK_ITERATION_LIMIT());
		KernelManager.getSingleton().addTask(gameTask);
		KernelManager.getSingleton().initAllTasks();

		run();
	}
}
